### ISSUE TITLE
ChefModernize/UnnecessaryDependsChef14 doesn't detect cookbooks with version constraints

### ISSUE THREAD

**Issue by damacus** [chef/cookstyle#323]
### What category of cop is this?:

Please try to select as few as possible.

- [x] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:

Depends on cookbook made obsolete by Chef 14

### What it would trigger on:

```ruby
# metadata.rb

depends 'build-essential', '>= 8.0.1'
```

### What it would autocorrect to if applicable

Remove the line

### Can We Help You Implement This?:

Yes please

**Comment by tas50** [comment]
I'm updating this to actually be a bug instead since we have this cop already, but it didn't support the version constraint.

**Comment by tas50** [comment]
Fixed by #325
